src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer.java

```diff
@@ -64,11 +64,19 @@ public void run(RunConfiguration config) throws AppEngineException {
       arguments.add(appYaml.toPath().toString());
     }
 
-    Map<String,String> env = Maps.newHashMap();
+    Map<String,String> environmentVariables = Maps.newHashMap();
     if (!Strings.isNullOrEmpty(config.getJavaHomeDir())) {
-      env.put("JAVA_HOME", config.getJavaHomeDir());
+      environmentVariables.put("JAVA_HOME", config.getJavaHomeDir());
     }
 
+    List<String> environment = config.getEnvironmentVariables();
+    if (environment != null) {
+      for (String nameValue : environment) {
+        String[] pair = nameValue.split("=");
```

Comment:
An environment variable value is allowed to have `=`. I agree with @chanseokoh — push the onus of splitting to the library consumers.